diff --git a/extlibs/gtest/SConscript b/extlibs/gtest/SConscript
index 4bcabc1..8a6f745 100644
--- a/extlibs/gtest/SConscript
+++ b/extlibs/gtest/SConscript
@@ -12,7 +12,7 @@ gtest_env = env.Clone()
 target_os = gtest_env.get('TARGET_OS')
 src_dir = gtest_env.get('SRC_DIR')
 
-targets_need_gtest = ['darwin','linux']
+targets_need_gtest = []
 gtest_dir      = os.path.join(src_dir, 'extlibs', 'gtest', 'gtest-1.7.0')
 gtest_lib_dir = os.path.join(gtest_dir,'lib')
 gtest_dotlib_dir =  os.path.join(gtest_lib_dir, '.libs')
