diff --git a/build_common/windows/SConscript b/build_common/windows/SConscript
index a6613b89d..04a34681f 100644
--- a/build_common/windows/SConscript
+++ b/build_common/windows/SConscript
@@ -189,7 +189,7 @@ if env['CC'] == 'cl':
         '/wd4232', '/wd4503', '/wd4512', '/wd4706'])
 
     # Enabling /W4 warnings globally for Windows builds.
-    env.AppendUnique(CCFLAGS=['/W4', '/WX'])
+    env.AppendUnique(CCFLAGS=['/W4'])
     env.AppendUnique(CCFLAGS=['/EHsc'])
 
     # Set release/debug flags
