diff --git a/resource/csdk/stack/octbstack_product.def b/resource/csdk/stack/octbstack_product.def
index 0d73f32e7..56021c559 100644
--- a/resource/csdk/stack/octbstack_product.def
+++ b/resource/csdk/stack/octbstack_product.def
@@ -14,7 +14,6 @@ convertTriggerEnumToString
 FindResourceByUri
 OCBindResource
 OCBindResourceHandler
-OCBindResourceInsToResource
 OCBindResourceInterfaceToResource
 OCBindResourceTypeToResource
 OCByteStringCopy
@@ -62,7 +61,6 @@ OCGetResourceInterfaceName
 OCGetResourceProperties
 OCGetResourceTypeName
 OCGetResourceUri
-OCGetResourceIns
 OCGetServerInstanceIDString
 OCGetSupportedEndpointTpsFlags
 OCInit
diff --git a/resource/csdk/stack/octbstack_rd_client.def b/resource/csdk/stack/octbstack_rd_client.def
index 0e322c976..5534a29b4 100644
--- a/resource/csdk/stack/octbstack_rd_client.def
+++ b/resource/csdk/stack/octbstack_rd_client.def
@@ -1,5 +1,7 @@
 ; APIs used when RD_MODE includes CLIENT
 
+OCBindResourceInsToResource
+OCGetResourceIns
 OCRDDelete
 OCRDDeleteWithDeviceId
 OCRDDiscover
diff --git a/resource/csdk/stack/octbstack_rd_server.def b/resource/csdk/stack/octbstack_rd_server.def
index 8cdba702f..c679c9536 100644
--- a/resource/csdk/stack/octbstack_rd_server.def
+++ b/resource/csdk/stack/octbstack_rd_server.def
@@ -1,5 +1,7 @@
 ; APIs used when RD_MODE includes SERVER
 
+OCBindResourceInsToResource
+OCGetResourceIns
 OCRDDatabaseInit
 OCRDDatabaseClose
 OCRDDatabaseDeleteResources
