diff --git a/resource/csdk/resource-directory/SConscript b/resource/csdk/resource-directory/SConscript
index f28a5f3db..7f7edcedb 100644
--- a/resource/csdk/resource-directory/SConscript
+++ b/resource/csdk/resource-directory/SConscript
@@ -54,7 +54,7 @@ if 'SERVER' in rd_mode:
     rd_env.AppendUnique(CPPDEFINES=['RD_SERVER'])
 
 rd_env.PrependUnique(
-    LIBS=['octbstack', 'oc', 'oc_logger', 'connectivity_abstraction'])
+    LIBS=['octbstack', 'oc_logger', 'connectivity_abstraction'])
 
 if target_os not in ['windows']:
     rd_env.AppendUnique(
@@ -80,7 +80,6 @@ else:
 ######################################################################
 RD_SRC_DIR = 'src/'
 rd_src_c = []
-rd_src_cpp = []
 
 if 'SERVER' in rd_mode:
     rd_src_c += [
@@ -95,19 +94,13 @@ if 'CLIENT' in rd_mode:
 
 rd_src_all = rd_src_c
 
-if target_os not in ['arduino', 'darwin', 'ios']:
-    rd_src_cpp += [RD_SRC_DIR + 'RDClient.cpp']
-    if 'CLIENT' in rd_mode:
-        rd_src_all += rd_src_cpp
-
 if target_os not in ['arduino', 'darwin', 'ios', 'msys_nt', 'windows']:
     rdsdk_shared = rd_env.SharedLibrary('resource_directory', rd_src_all)
     rdsdk_static = rd_env.StaticLibrary('resource_directory', rd_src_all)
     rdsdk = Flatten([rdsdk_static, rdsdk_shared])
 elif target_os in ['msys_nt', 'windows']:
     rdsdk_c = rd_env.StaticLibrary('resource_directory_internal', rd_src_c)
-    rdsdk_cpp = rd_env.StaticLibrary('resource_directory', rd_src_cpp)
-    rdsdk = Flatten([rdsdk_c, rdsdk_cpp])
+    rdsdk = Flatten([rdsdk_c])
 else:
     rdsdk = rd_env.StaticLibrary('resource_directory', rd_src_all)
 
